src/main/java/tc/oc/pgm/prefix/ConfigPrefixProvider.java

```diff
@@ -0,0 +1,46 @@
+package tc.oc.pgm.prefix;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.UUID;
+import org.bukkit.Bukkit;
+import tc.oc.pgm.Config.Prefixes;
+import tc.oc.pgm.Config.Prefixes.Prefix;
+
+public class ConfigPrefixProvider implements PrefixProvider {
+
+  private final Map<UUID, String> prefixMap = new HashMap<UUID, String>();
+
+  @Override
+  public void setPrefix(UUID uuid, String prefix) {
+    Bukkit.getPluginManager()
+        .callEvent(new PrefixChangeEvent(uuid, prefixMap.put(uuid, prefix), prefix));
+  }
+
+  @Override
+  public String getPrefix(UUID uuid) {
+    if (!prefixMap.containsKey(uuid)) {
+      setPrefix(uuid, getPrefixFromConfig(uuid));
+    }
+    return prefixMap.get(uuid);
+  }
+
+  @Override
+  public void removePlayer(UUID uuid) {
+    prefixMap.remove(uuid);
+  }
+
+  private String getPrefixFromConfig(UUID uuid) {
+    if (!Bukkit.getPlayer(uuid).isOnline()) {
+      return "";
+    }
```

Comment:
Just because a player isn't online doesn't mean they can't have flairs, right? Does bukkit not allow getting the permissions of offline players?

Reply:
As far as I can see, it doesn't. Generally you can only obtain a Player object using the Bukkit API for a player who is online; I think Bukkit#getPlayer(UUID) usually returns null for offline players (and so this should probably be changed to reflect that).

For offline players, you need to get an OfflinePlayer, which doesn't extend Permissible and so doesn't have permissions. Furthermore, permissions aren't attached to players until they have joined, and attachments should be removed when they disconnect.